hydra-test-utils/src/Test/Hydra/Prelude.hs

```diff
@@ -97,6 +101,30 @@ failure :: (HasCallStack, MonadThrow m) => String -> m a
 failure msg =
   throwIO (HUnitFailure location $ Reason msg)
 
+orElse :: (MonadCatch m) => m a -> m a -> m a
+orElse m n =
+  m `catch` \(_ :: HUnitFailure) -> n
+
+-- XXX: shamelessly stolen from https://hackage.haskell.org/package/exceptions-0.10.7/docs/src/Control.Monad.Catch.html#MonadCatch
+-- because these instances are not implemented in io-classes
+instance MonadMask m => MonadThrow (StrictS.StateT s m) where
+  throwIO e = lift $ throwIO e
+
+instance (MonadMask m) => MonadCatch (StrictS.StateT s m) where
+  catch = StrictS.liftCatch catch
+
+instance MonadMask m => MonadMask (StrictS.StateT s m) where
+
+  mask a = StrictS.StateT $ \s -> mask $ \u -> StrictS.runStateT (a $ q u) s
+    where q :: (m (a, s) -> m (a, s)) -> StrictS.StateT s m a -> StrictS.StateT s m a
+          q u (StrictS.StateT b) = StrictS.StateT (u . b)
+
+  uninterruptibleMask a =
+    StrictS.StateT $ \s -> uninterruptibleMask $ \u -> StrictS.runStateT (a $ q u) s
+      where q :: (m (a, s) -> m (a, s)) -> StrictS.StateT s m a -> StrictS.StateT s m a
+            q u (StrictS.StateT b) = StrictS.StateT (u . b)
+
+
```

Comment:
We could remove this if we collect all the effects before checking stuff in the tests